### ISSUE TITLE
calc_Huntley2006() ... potentially massive performance regression

### ISSUE THREAD

**Issue by RLumSK** [R-Lum/Luminescence#258]
**Describe the bug**
When running the example dataset with the default number of `n.MC = 100000`, the function takes
ages to complete the calculation. `n.MC = 10000` takes just a few seconds. 

I did a profiling and it seems the one `mapply()` call causes a massive lose of performance, which we did not observe in the past (see screenshot) 

*Notes*
* the `n.MC = 100000` came in with 25c24c11765c4464ca100f5a4430e4a14513b811
* the mapply() with 6953f5226fd379e341434443494bc9ac49b283d5

Means, its in there a long time and it had certainly worked before as expected.

**Please add a minimalistic reproducible example**

This example is from the package manual. 
```r
library(Luminescence)

## Load example data (sample UNIL/NB123, see ?ExampleData.Fading)
data("ExampleData.Fading", envir = environment())

## (1) Set all relevant parameters
# a. fading measurement data (IR50)
fading_data <- ExampleData.Fading$fading.data$IR50

# b. Dose response curve data
data <- ExampleData.Fading$equivalentDose.data$IR50

## (2) Define required function parameters
ddot <- c(7.00, 0.004)
readerDdot <- c(0.134, 0.0067)

# Analyse fading measurement and get an estimate of rho'.
# Note that the RLum.Results object can be directly used for further processing.
# The number of MC runs is reduced for this example
rhop <- analyse_FadingMeasurement(fading_data, plot = TRUE, verbose = FALSE, n.MC = 10)

## (3) Apply the Kars et al. (2008) model to the data
kars <- calc_Huntley2006(
 data = data,
 rhop = rhop,
 ddot = ddot,
 readerDdot = readerDdot,
 n.MC = 100000)
```

I have played around with `mapply()` and it seems that

```{r}
mapply(rep, times = 1:1e5, x = 4:1)
```

is finished in seconds, while

```{r}
mapply(rep, times = 1:1e6, x = 4:1)
```
took ages and before finally reaching the vector maximum of 100 GB on my machine. 

* R 4.3.3 (0.9.25): Runtime 6.72017 mins
* R 4.4.1 (0.9.25): Runtime 6.600699 mins 

**Expected behaviour**

Execution in reasonable time. We did not have the problem in the past. 

**Screenshots**

<img width="1335" alt="Screenshot 2024-09-20 at 20 20 01" src="https://github.com/user-attachments/assets/7b6aa3bf-cf53-4abe-b845-06b5b386a217">

**Comment by RLumSK** [comment]
I will close this for the moment after reducing the default. I could not get any better numbers with three R versions, so for the moment it is not worth investing it further.

**Comment by mcol** [comment]
Just a note: `mapply()` is probably unrelated, the massive slow-down is just due to the amount of memory being allocated in those examples. 
```R
res <- mapply(rep, times = 1:1e5, x = 4:1)
#                                  Type     Size   Rows Columns
# res                              list  18.6 Gb 100000      NA
```

**Comment by mcol** [comment]
Yesterday I ran out of time to write this, but I ended up giving a better look at how we use that `mapply()`, and I've found ways to speed it up:
- step 1: some computations can be moved outside of `mapply()` because they are independent of the loop variables
- step 2: a minor speed-up can be obtained by simplifying algebraically the terms for `rho`, `r` and `tau`, since currently some of the operations done for `rho` and `r` are then undone in `tau` 

Step 3 can bring another quite large speed up, but I'd like confirmation that I'm not pushing things too far. Currently we generate `rprime` like this:
https://github.com/R-Lum/Luminescence/blob/03874ed2f1e0078f575aedad62aa97fade5ff685/R/calc_Huntley2006.R#L677
I'm wondering why for say `n_MC = 100000` we would need so many points between 0.01 and 5: it sounds to me that once we've got a fine enough partitioning of this interval (say with 1000 points), we don't gain much by dividing it further.

For example, this are the differences for `n_MC = 10000` between using 1000 points (14s) and 10000 points (17s) in `rprime`:
```
               nN_SS nN_SS.error
- old[1, ] 0.3678342  0.07874376
+ new[1, ] 0.3678340  0.07874367

`old$nN_SS`: 0.36783422
`new$nN_SS`: 0.36783402

`old$nN_SS.error`: 0.078743759
`new$nN_SS.error`: 0.078743673
```
Speed gains are much larger as `n_MC` grows. For `n_MC = 50000` here are timings I get in seconds:
```
master: 230.200
step 1: 149.490
step 2: 137.351
step 3:  61.568
```
